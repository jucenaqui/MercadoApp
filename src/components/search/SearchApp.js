import React, { useState } from 'react';

import './SearchApp.scss';

const SearchApp = ( { history } ) => {

    const [inputValue, setInputValue] = useState('');
    
    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if ( inputValue.trim().length > 2 ) {
            history.push(`/items?search=${inputValue}`);
        }
    }
    return (
        <form onSubmit={ handleSubmit } className="search-form">
            <input 
            type="text"
            value={ inputValue }
            onChange={handleInputChange}
            />
        </form>
    )
}

export default SearchApp;