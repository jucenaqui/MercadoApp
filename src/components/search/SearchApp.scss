.search-form {
    width: 80%;
    padding: 5px;

    input {
        background-color: #fff;
        border: none;
        border-radius: 2px;
        outline: none;
        padding: 0 10px;
        box-shadow: 0 1px 2px 0 rgba(0 ,0 ,0 , 0.12);
        box-sizing: border-box;
        height: 30px;
    }
}