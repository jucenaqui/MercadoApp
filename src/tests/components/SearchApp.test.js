import { shallow } from "enzyme";
import history from 'history'
import SearchApp from "../../components/search/SearchApp";

describe('Test to <SearchApp />', () => {

    const wrapper = shallow(<SearchApp history={ history } />);

    test('should show the component succesfull', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('should change input text', () => {
        const input = wrapper.find('input');
        const value = 'nevera';
        input.simulate('change', {target:{value}});
    });

})
