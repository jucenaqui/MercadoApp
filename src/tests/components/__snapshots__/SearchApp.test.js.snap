// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to <SearchApp /> should show the component succesfull 1`] = `
<form
  className="search-form"
  onSubmit={[Function]}
>
  <div
    className="input-group"
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <FaSearch />
  </div>
</form>
`;
